libpod/container_api.go

```diff
@@ -834,3 +840,181 @@ func (c *Container) Refresh(ctx context.Context) error {
+	if err := c.cleanup(); err != nil {
+		return err
+	}
+
+	if keep {
+		return c.save()
+	}
+
+	// Remove log file
+	os.Remove(filepath.Join(c.bundlePath(), "dump.log"))
+	// Remove statistic file
+	os.Remove(filepath.Join(c.bundlePath(), "stats-dump"))
+
+	return c.save()
+}
+
+// Restore restores a container
+func (c *Container) Restore(keep bool) (err error) {
+	if !c.batched {
+		c.lock.Lock()
+		defer c.lock.Unlock()
+
+		if err := c.syncContainer(); err != nil {
+			return err
+		}
+	}
+
+	if c.state.State == ContainerStateRunning {
+		return errors.Wrapf(ErrCtrStateInvalid, "%q is running, cannot restore", c.state.State)
+	}
+
+	// Read network configuration from checkpoint
+	// Currently only one interface with one IP is supported.
+	networkStatusFile, err := os.Open(filepath.Join(c.bundlePath(), "network.status"))
+	if err == nil {
+		// The file with the network.status does exist. Let's restore the
+		// container with the same IP address as during checkpointing.
+		defer networkStatusFile.Close()
+		var networkStatus []*cnitypes.Result
+		networkJSON, err := ioutil.ReadAll(networkStatusFile)
+		if err != nil {
+			return err
+		}
+		json.Unmarshal(networkJSON, &networkStatus)
+		// Take the first IP address
+		var IP net.IP
+		if len(networkStatus) > 0 {
+			if len(networkStatus[0].IPs) > 0 {
+				IP = networkStatus[0].IPs[0].Address.IP
+			}
+		}
+		if IP != nil {
+			env := fmt.Sprintf("IP=%s", IP)
+			// Tell CNI which IP address we want.
+			os.Setenv("CNI_ARGS", env)
+			logrus.Debugf("Restoring container with %s", env)
+		}
+	}
+
+	if err := c.prepare(); err != nil {
+		return err
+	}
+	defer func() {
+		if err != nil {
+			if err2 := c.cleanup(); err2 != nil {
+				logrus.Errorf("error cleaning up container %s: %v", c.ID(), err2)
+			}
+		}
+	}()
+
+	// CNI_ARGS was used to request a certain IP address. Unconditionally remove it.
+	os.Unsetenv("CNI_ARGS")
+
+	// Read config
+	jsonPath := filepath.Join(c.bundlePath(), "config.json")
+	logrus.Debugf("generate.NewFromFile at %v", jsonPath)
+	g, err := generate.NewFromFile(jsonPath)
+	if err != nil {
+		logrus.Debugf("generate.NewFromFile failed with %v", err)
+		return err
+	}
+
+	// We want to have the same network namespace as before.
+	if c.config.CreateNetNS {
+		g.AddOrReplaceLinuxNamespace(spec.NetworkNamespace, c.state.NetNS.Path())
+	}
+
+	// Save the OCI spec to disk
+	if err := c.saveSpec(g.Spec()); err != nil {
+		return err
+	}
+
+	// Copy /etc/resolv.conf to the container's rundir
+	_, err = c.generateResolvConf()
+	if err != nil {
+		return err
+	}
+
+	// Copy /etc/hosts to the container's rundir
+	_, err = c.generateHosts()
+	if err != nil {
+		return errors.Wrapf(err, "unable to copy /etc/hosts to container space")
+	}
+
+	// This needs to be deleted so that the restore works correctly.
+	// This also happens in Checkpoint()
+	// Not sure where is the right place for this.
+	// If the container is stopped via runc, it needs to be here for a successful restart.
```

Comment:
I think what you're looking for is `removeConmonFiles()`

Reply:
Good to know it exists. Thanks. Will use it.